grpc-examples/routeguide/server/utils.bal

```
@@ -0,0 +1,91 @@
+RouteNote[] ROUTE_NOTES = [];
+
+function sendRouteNotesFromLocation(RouteGuideRouteNoteCaller caller, Point location) {
+    lock {
+        foreach RouteNote note in ROUTE_NOTES {
+            if note.location == location {
+                grpc:Error? e = caller->sendRouteNote(note);
+                if e is grpc:Error {
+                    grpc:Error? sendErr = caller->sendError(e);
+                }
+            }
+        }
+    }
+}
+
+function addRouteNotes(stream<RouteNote, grpc:Error?> clientStream) returns error? {
+    check clientStream.forEach(function(RouteNote note) {
+        lock {
+            ROUTE_NOTES.push(note);
+        }
+    });
+}
+
+isolated function toRadians(float f) returns float {
+    return f * 'float:PI / 180.0;
+}
+
+function calculateDistance(Point p1, Point p2) returns int {
+    float cordFactor = 10000000; // 1x(10^7) OR 1e7
+    float R = 6371000; // Earth radius in metres
+    float lat1 = toRadians(<float>p1.latitude / cordFactor);
+    float lat2 = toRadians(<float>p2.latitude / cordFactor);
+    float lng1 = toRadians(<float>p1.longitude / cordFactor);
+    float lng2 = toRadians(<float>p2.longitude / cordFactor);
+    float dlat = lat2 - lat1;
+    float dlng = lng2 - lng1;
+
+    float a = 'float:sin(dlat / 2.0) * 'float:sin(dlat / 2.0) + 'float:cos(lat1) * 'float:cos(lat2) * 'float:sin(dlng / 2.0) * 'float:sin(dlng / 2.0);
+    float c = 2.0 * 'float:atan2('float:sqrt(a), 'float:sqrt(1.0 - a));
+    float distance = R * c;
+    return <int>distance;
+}
+
+
+isolated function pointExistsInFeatures(Feature[] features, Point point) returns boolean {
+    foreach Feature feature in features {
+        if feature.location == point {
+            return true;
+        }
+    }
+    return false;
+}
+
+isolated function featureFromPoint(Point point) returns Feature?|error {
+    foreach Feature feature in FEATURES {
+        if feature.location == point {
+            return feature;
+        }
+    }
+    return ();
+}
```

we don't need to return an error here. is it possible to we move this logic to remote function

---

Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/320/commits/7a57af9701dc7803c8d433a610e7e9f62404a75f